geomesa-hbase/geomesa-hbase-datastore/src/main/java/org/locationtech/geomesa/hbase/filters/JSimpleFeatureFilter.java

```diff
@@ -65,17 +78,20 @@ public ReturnCode filterKeyValue(Cell v) throws IOException {
     }
 
     private static class CQLFilter extends AbstractFilter {
-        KryoBufferSimpleFeature reusable;
-        CQLFilter(String sftString, String filter) {
-            super(sftString, filter);
-            reusable = IteratorCache.serializer(sftString, SerializationOptions.withoutId()).getReusableFeature();
+        CQLFilter(String sftString, String filter, String transform, String transformSchema) {
+            super(sftString, filter, transform, transformSchema);
         }
 
         @Override
         public ReturnCode filterKeyValue(Cell v) throws IOException {
             try {
                 log.trace("cell = {}", v);
-                reusable.setBuffer(v.getValueArray(), v.getValueOffset(), v.getValueLength());
+                if (hasTransform) {
```

Comment:
i think we should revert this change here - if there are issues with filtering we can just always clone the value until we figure it out, but it doesn't make sense to have any logic concerning transforms in this method.